lib/mlpatches/modules/subprocess.py

```
@@ -0,0 +1,301 @@
+# -*- coding: utf-8 -*-
+"""Subprocess management"""
+# info: parts of this module has been copied from the original subprocess-module
+
+# === NotImplemented ===
+# - the following args are always ignored:
+# - executable
+# - shell
+# - startupinfo
+# - preexec_fn
+# - bufsize
+# - creationflags
+# - close_fds
+# Other implemention differences:
+# - terminate() is just a kill()
+# - send_signal() always call kill()
+# - communicate() splits data in chunks of 4096 bytes before sending
+# - poll() never returns the signal that killed the process
+# - pid is the job_id
+
+import os
+import select
+import time
+import sys
+from mlpatches import base, l2c
+from six import integer_types, string_types
+
+_stash = base._stash
+list2cmdline = l2c.list2cmdline
+
+# constants
+try:
+    # setup MAXFD. we don't require this, but other scripts may expect this value to be present
+    MAXFD = os.sysconf("SC_OPEN_MAX")
+except:
```

Please do not add new instances of bare except as discussed in PEP8.
* https://docs.astral.sh/ruff/rules/bare-except
* https://peps.python.org/pep-0008/#programming-recommendations

---

This code in general was rolled back to original master. There are was a lot of bare excepts